packages/node-core/src/indexer/storeModelProvider/global/global.ts

```diff
@@ -59,41 +61,65 @@ export class PlainGlobalModel implements IGlobalData {
   async setGlobalRewindLock(rewindTimestamp: Date): Promise<{lockTimestamp: Date}> {
     const tx = await this.sequelize.transaction();
     const {chainsCount, currentChain} = await this.getChainsInfo(tx);
-
-    try {
-      let lockTimestamp = new Date(0);
-      const [affectedCount] = await this.model.update(
-        {
-          status: MultiChainRewindStatus.Incomplete,
-          rewindTimestamp,
-          initiator: false,
-        },
-        {
-          where: {
-            [Op.or]: [{status: MultiChainRewindStatus.Normal}, {rewindTimestamp: {[Op.gt]: rewindTimestamp}}],
+    if (chainsCount) {
+      // Exist rewind task
+      try {
+        let lockTimestamp = new Date(0);
+        const [affectedCount] = await this.model.update(
+          {
+            status: MultiChainRewindStatus.Incomplete,
+            rewindTimestamp,
+            initiator: false,
           },
-          transaction: tx,
+          {
+            where: {
+              [Op.or]: [{status: MultiChainRewindStatus.Normal}, {rewindTimestamp: {[Op.gt]: rewindTimestamp}}],
+            },
+            transaction: tx,
+          }
+        );
+
+        assert(
+          affectedCount === 0 || affectedCount === chainsCount,
+          `Set global rewind lock failed, chainId: ${this.chainId}, rewindTimestamp: ${rewindTimestamp}`
+        );
+
+        if (affectedCount === chainsCount) {
+          lockTimestamp = rewindTimestamp;
+          await this.model.update({initiator: true}, {where: {chainId: this.chainId}, transaction: tx});
         }
-      );
-
-      assert(
-        affectedCount === 0 || affectedCount === chainsCount,
-        `Set global rewind lock failed, chainId: ${this.chainId}, rewindTimestamp: ${rewindTimestamp}`
-      );
 
-      if (affectedCount === chainsCount) {
-        lockTimestamp = rewindTimestamp;
-        await this.model.update({initiator: true}, {where: {chainId: this.chainId}, transaction: tx});
+        await tx.commit();
+        return {lockTimestamp};
+      } catch (e: any) {
+        logger.error(
+          `setGlobalRewindLock failed chainId: ${this.chainId}, rewindTimestamp: ${rewindTimestamp}, errorMsg: ${e.message}`
+        );
+        await tx.rollback();
+        throw e;
+      }
+    } else {
+      const chainIds = await this.getChainIdsFromMetadata();
```

Comment:
I think this will want to use the transaction to avoid a new chain being created between this and bulk create